ambry-filesystem/src/main/resources/AmbryFS.ddl

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright 2023 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+
+/* CREATE DATABASE ambryfs; */
+
+CREATE TABLE IF NOT EXISTS DirectoryInfo (
+    account_id      int           NOT NULL,
+    container_id    int           NOT NULL,
+    parent_dir_id   binary(16)    NOT NULL,
+    resource_name   varchar(128)  NOT NULL,
+    resource_type   byte          NOT NULL,
+    resource_id     binary(16)    NOT NULL,
+    created_ts      datetime(6)   NOT NULL,
+    last_updated_ts datetime(6)   NOT NULL,
+    deleted_ts      datetime(6)   DEFAULT NULL,
+
+    PRIMARY KEY (account_id, container_id, parent_dir_id, resource_name)
```

Comment:
I was hoping we could follow this model which is more k/v friendly: https://docs.google.com/document/d/1nXEXDhorJf4ylme9Eb8wqgs1ket5igTDgdYAhvoUp68/edit#bookmark=id.9ly0e83bv2p

Based on this model, a directory is keyed with parent_object_id/directory_name.

Reply:
I don't think we have to worry about mysql database schema isn't kv friendly. If later we start using a simple key-value store, we would have a different implementation.

Also I think the account and container is useful to be part of the primary key in the table, so we can quickly find how many directories and files created by an account and a container.